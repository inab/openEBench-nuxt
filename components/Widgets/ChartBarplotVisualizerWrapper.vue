<template>
	<div
		:data-id="id"
		class="benchmarkingChart_bars"
		:data-data="JSON.stringify(data)"
		:data-metric-name="metricName"
	/>
</template>

<script>
// eslint-disable-next-line camelcase
import { load_bars_visualization } from '@inb/oeb-chart-barplot';

export default {
	props: {
		id: {
			type: String,
			required: true,
		},
		data: {
			type: Array,
			required: true,
		},
		metricName: {
			type: String,
			required: true,
		},
	},
	data() {
		return {
			apiUrl: this.$config
				? this.$config.SCIENTIFIC_SERVICE_URL + '/graphql'
				: 'https://dev-openebench.bsc.es/api/scientific/graphql',
		};
	},
	mounted() {
		this.loadVisualization();
	},
	methods: {
		loadVisualization() {
			try {
				load_bars_visualization();
			} catch (error) {}
		},
	},
};
</script>

<style lang="scss">
.benchmarkingTable_bars {
	background-color: #fff;
	width: 28vw;
}
</style>
