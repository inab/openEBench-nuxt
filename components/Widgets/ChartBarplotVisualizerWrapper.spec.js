import { shallowMount } from '@vue/test-utils';
import ChartBarplotVisualizerWrapper from './ChartBarplotVisualizerWrapper.vue';

jest.mock('@inb/oeb-chart-barplot');
// eslint-disable-next-line
import { load_bars_visualization } from '@inb/oeb-chart-barplot';

const factory = () => {
	return shallowMount(ChartBarplotVisualizerWrapper, {
		...createComponentMocks({}),
		propsData,
	});
};

const propsData = {
	id: 'someid',
	data: [
		{
			toolname: 'group01',
			mode: 'lines',
			type: 'bar-plot',
			metric_value: 0.936,
		},
		{
			toolname: 'group02',
			mode: 'lines',
			type: 'bar-plot',
			metric_value: 0.9279,
		},
	],
	metricName: 'Metric Name',
};

describe('ChartBarplotVisualizerWrapper', () => {
	afterEach(() => {
		jest.clearAllMocks();
	});

	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory();
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('is calls load_bars_visualization after mount', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
		expect(load_bars_visualization).toHaveBeenCalled();
	});

	it('sets the apiUrl according to SCIENTIFIC_SERVICE_URL env var', () => {
		const wrapper = factory();
		expect(wrapper.vm.$data.apiUrl).toBe(
			'https://jest-openebench.bsc.es/api/scientific/graphql'
		);
	});
});
