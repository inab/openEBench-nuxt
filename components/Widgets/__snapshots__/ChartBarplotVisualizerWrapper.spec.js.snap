// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartBarplotVisualizerWrapper should match snapshot 1`] = `"<div data-id=\\"someid\\" data-data=\\"[{&quot;toolname&quot;:&quot;group01&quot;,&quot;mode&quot;:&quot;lines&quot;,&quot;type&quot;:&quot;bar-plot&quot;,&quot;metric_value&quot;:0.936},{&quot;toolname&quot;:&quot;group02&quot;,&quot;mode&quot;:&quot;lines&quot;,&quot;type&quot;:&quot;bar-plot&quot;,&quot;metric_value&quot;:0.9279}]\\" data-metric-name=\\"Metric Name\\" class=\\"benchmarkingChart_bars\\"></div>"`;
